Makefile

```diff
@@ -1,43 +1,41 @@
 #
-# Makefile for phoenix-rtos-devices
+# Makefile for phoenix-rtos-tests
 #
 # Copyright 2018, 2019 Phoenix Systems
 #
 # %LICENSE%
 #
 
-SIL ?= @
-MAKEFLAGS += --no-print-directory
-
-TARGET ?= ia32-generic
-
 include ../phoenix-rtos-build/Makefile.common
+# FIXME: this include should be done by Makefile.common
 include ../phoenix-rtos-build/Makefile.$(TARGET_SUFF)
 
-INSTALL_PROGS :=
+.DEFAULT_GOAL := all
 
-.PHONY: clean
-clean:
-	@echo "rm -rf $(BUILD_DIR)"
-
-ifneq ($(filter clean,$(MAKECMDGOALS)),)
-	$(shell rm -rf $(BUILD_DIR))
+ifneq ($(filter %clean,$(MAKECMDGOALS)),)
+$(info cleaning targets, make parallelism disabled)
+.NOTPARALLEL:
 endif
 
-T1 := $(filter-out clean all install,$(MAKECMDGOALS))
-ifneq ($(T1),)
-	include $(T1)/Makefile
-.PHONY: $(T1)
-$(T1): all
-else
-	include Makefile.$(TARGET_SUFF)
-endif
+# shortcut for unity dependence
+define add_unity_test
+$(call add_test,$(1),$(2),unity)
+endef
 
-PREFIX_INSTALL = $(PREFIX_ROOTFS)bin/
+# default path for the programs to be installed in rootfs
+DEFAULT_INSTALL_PATH := /bin
 
-$(addprefix $(PREFIX_INSTALL), $(INSTALL_PROGS)): $(PREFIX_INSTALL)% : $(PREFIX_PROG_STRIPPED)%
-	$(INSTALL_FS)
+# read out all components
+ALL_MAKES := $(shell find -mindepth 2 -name Makefile)
+include $(ALL_MAKES)
 
-all: $(addprefix $(PREFIX_PROG), $(INSTALL_PROGS))
+# by default compile all tests, but allow custom values on per-TARGET_FAMILY basys
+# TODO: prepare tool which will read YAML files and return the components which should be compiled for a given platform
```

Comment:
@jsarzy - idea for the future: 
write separate tool (which uses some classes from the runner) to compute the subset of components / tests to be run on a given target and use it here instead of keeping the per-target configuration on two places (YAML files and Makefiles).